src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -279,14 +279,28 @@ private void WriteGetVariants(RestClientMethod method)
 
         private void WriteListVariants()
         {
-            if (_resourceContainer.ListMethod != null)
+            if (_resourceContainer.ListMethods != null)
```

Comment:
this null check is probably not needed since we can rely on foreach'ing over an empty list?

Reply:
Since this property is nullable, this check is needed in order to access _resourceContainer.ListMethod property.